sympy/matrices/matrices.py

```diff
@@ -378,6 +380,18 @@ def diagonalize(self, reals_only=False, sort=False, normalize=False):
         return _diagonalize(self, reals_only=reals_only, sort=sort,
                 normalize=normalize)
 
+    def _bidiag_hholder(self):
+        return _eval_bidiag_hholder(self)
+
+    def bidiagonalize(self, upper=True):
+        return _bidiagonalize(self, upper=upper)
+
+    def bidiagonal_decomposition(self,upper=True):
+        return _bidiagonal_decomposition(self,upper=upper)
+
+    def bidiagonal_decmp_hholder(self):
```

Comment:
Why is this needed?

Reply:
No particular reason, but while reading Sympy code earlier, I had seen multiple functions, like

```python3
_eval_det_lu,
_eval_det_bareiss
_eval_det_berkowitz
```
along with
```python
det
```

so I did the same way.
Should I remove
```python
bidiagonal_decmp_hholder
```

Reply:
`_eval_det` stuff are private methods while `bidiagonal_decmp_hholder` is a public method. So we should remove the redundant public method.

Reply:
Okay, removed `bidiagonal_decmp_hholder`